### ISSUE TITLE
UIRequiredDeviceCapabilities must also be present in frameworks' info.plist when container app is set to arm64 to avoid ERROR ITMS-90502

### ISSUE THREAD

**Issue by dalexsoto** [xamarin/xamarin-macios#3581]
## Description

When container app's `info.plist` has `UIRequiredDeviceCapabilities` set to `arm64` only i.e.

```xml
<key>UIRequiredDeviceCapabilities</key>
<array>
    <string>arm64</string>
</array>
```

all frameworks' `info.plist` inside `Frameworks` directory should also contain `UIRequiredDeviceCapabilities` set to `arm64` else you will get the following error when submitting your app to iTunes Connect:

> ERROR ITMS-90502: "Invalid Bundle. Apps that only contain the arm64 slice must also have 'arm64' in the list of UIRequiredDeviceCapabilities in Info.plist.

References:
* https://github.com/CocoaPods/CocoaPods/issues/4514
* https://forums.developer.apple.com/thread/25411
* https://forums.developer.apple.com/thread/84456

### Steps to Reproduce
1. Create a new iOS app
2. Set `UIRequiredDeviceCapabilities` to `arm64` only inside app's info.plist.
3. Add a framework like [SkiaSharp](https://www.nuget.org/packages/SkiaSharp) and add a simple code reference to it somewhere in the app, something simple like `var path = new SkiaSharp.SKPath ();` should do.
4. Build your app for release.
5. Try to submit using iTunes connect

### Expected Behavior
All Frameworks' `info.plist` should contain the following entry: 

```xml
<key>UIRequiredDeviceCapabilities</key>
<array>
    <string>arm64</string>
</array>
```

if container app info.plist's `UIRequiredDeviceCapabilities` is set to only `arm64`  also app should upload successfully using app uploader.

### Actual Behavior
ERROR ITMS-90502: "Invalid Bundle. Apps that only contain the arm64 slice must also have 'arm64' in the list of UIRequiredDeviceCapabilities in Info.plist.

### Environment
```
=== Visual Studio Enterprise 2017 for Mac (Preview) ===

Version 7.4 Preview (7.4 build 985)
Installation UUID: 4e014908-906a-4d2b-9aed-1d598a6d71cf
Runtime:
	Mono 5.10.0.105 (2017-12/4b1745401d5) (64-bit)
	GTK+ 2.24.23 (Raleigh theme)

	Package version: 510000105

=== Apple Developer Tools ===

Xcode 9.2 (13772)
Build 9C40b

=== Xamarin.Mac ===

Version: 4.5.0.0 (Visual Studio Enterprise)

=== Xamarin.iOS ===

Version: 11.11.0.0 (Visual Studio Enterprise)
Hash: 0dd292f
Branch: master
Build date: 2018-02-21 19:51:18-0600

=== Operating System ===

Mac OS X 10.13.3
Darwin 17.4.0 Darwin Kernel Version 17.4.0
    Sun Dec 17 09:19:54 PST 2017
    root:xnu-4570.41.2~1/RELEASE_X86_64 x86_64
```

## Workaround

Place the following code into your main app `csproj` right before the closing `</Project>` tag:

```xml
<Target Name="BeforeCodesign" DependsOnTargets="_CollectFrameworks">
    <Exec Command="xcrun plutil -replace UIRequiredDeviceCapabilities -json '[ &quot;arm64&quot; ]' /%(_Frameworks.Directory)/Info.plist" />
</Target>
```

**Comment by dalexsoto** [comment]
We have to be careful with the handling of this setting because once is set and uploaded to the AppStore it can't be changed -> https://developer.apple.com/library/content/qa/qa1623/_index.html

> Important: Because you can't add `UIRequiredDeviceCapabilities` restrictions after an app is in the store, be sure to plan ahead by choosing requirements you will be comfortable supporting indefinitely going forward.

**Comment by scastria** [comment]
A lot of people have said that all dependencies must have this restriction, but I have found that not to be true.  I have successfully uploaded an app with only my app and its extension with arm64, but none of its dependencies.  You also can find lots of posts saying that after a rebuild and re-upload with no changes, it worked.  They then think Apple fixed something.  So I am not sure if anyone knows what is really going on here.
